returnn/tf/util/data.py

```diff
@@ -295,6 +295,25 @@ def get_uniq_collection(cls, tags, is_equal_opts=None):
         res.append(tag)
     return res
 
+  def get_dim_value(self):
+    """
+    Infers the dim this axis should have if unbroadcasted.
+    If `self.src_data` has a placeholder, will use the shape from there.
+    Otherwise, uses `self.dimension` (if static) or `self.dyn_size` (if dynamic).
+    :rtype: int|Tensor
+    """
+    if self.dimension is not None:
+      return self.dimension
+    if self.kind == self.Types.Batch:
+      from returnn.tf.layers.base import LayerBase
+      return LayerBase.get_recent_layer().get_batch_dim()
+    if self.src_data is not None and self.src_axis is not None and self.src_data.placeholder is not None:
+      from returnn.tf.util.basic import get_shape
+      return get_shape(self.src_data.placeholder)[self.src_axis]
+    if self.dyn_size is not None:
+      return tf.math.reduce_max(self.dyn_size)
```

Comment:
Btw, since when is it `tf.math.reduce_max`? Already since TF 1.8? (I'm also not sure what minimum TF version we want to support. If all tests are passing, and noone complains, we can just leave it.)

Reply:
I don't know. Tests seem to pass though :D